src/Features/Core/Portable/Diagnostics/BuildOnlyDiagnosticsService.cs

```diff
@@ -103,43 +91,24 @@ private void ClearDiagnostics(Project? project)
 
             lock (_gate)
             {
-                _projectDiagnostics.Remove(project.Id);
                 foreach (var documentId in project.DocumentIds)
                     _documentDiagnostics.Remove(documentId);
             }
         }
 
-        public void ClearBuildOnlyDiagnostics(Solution solution, ProjectId? projectId, DocumentId? documentId)
+        public void ClearBuildOnlyDiagnostics(Project project, DocumentId? documentId)
```

Comment:
passing in a project-id here meant 'clear out project diags, and all diags for any document in this project'.  So, maintaining that logic here.